core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -376,7 +381,9 @@ class LogManager(logDirs: Seq[File],
                 s"($currentNumLoaded/${logsToLoad.length} loaded in $logDirAbsolutePath)")
             } catch {
               case e: IOException =>
-                offlineDirs.add((logDirAbsolutePath, e))
+                handleIOException(logDirAbsolutePath, e)
+              case e: KafkaStorageException if e.getCause.isInstanceOf[IOException] =>
+                // KafkaStorageException might be thrown, ex: during writing LeaderEpochFileCache
```

Comment:
When converting IOException to KafkaStorageException, we already add an error log. So, there is probably no need to log it below again. Could we also add a comment why we can just let go of KafkaStorageException?

Reply:
Yes, good suggestion.